app/models/communitarian/user.rb

```diff
@@ -0,0 +1,9 @@
+# frozen_string_literal: true
+
+module Communitarian
+  refine User do
+    def posted_recently?(topic_id)
+      self.posts.where(topic_id: topic_id, created_at: 5.minutes.ago..Time.current).exists?
```

Comment:
I think that this condition should work a little bit faster
```suggestion
      self.posts.where(topic_id: topic_id).where("created_at > ?", 5.minutes.ago).exists?
```